hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieHFileDataBlock.java

```diff
@@ -122,9 +118,9 @@ public HoodieLogBlockType getBlockType() {
       if (useIntegerKey) {
         recordKey = String.format("%" + keySize + "s", key++);
       } else {
-        recordKey = record.get(keyField.pos()).toString();
+        recordKey = record.get(schemaKeyField.pos()).toString();
```

Comment:
why not just look up by name? Don't understand this need for positional lookups to begin with?

Reply:
IndexedRecord supports only get by index. Not by name.